bundles/test/org.eclipse.smarthome.magic/src/main/java/org/eclipse/smarthome/magic/binding/internal/automation/modules/MagicActionProvider.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.magic.binding.internal.automation.modules;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.smarthome.automation.ActionProvider;
+import org.eclipse.smarthome.automation.annotation.ActionInput;
+import org.eclipse.smarthome.automation.annotation.ActionOutput;
+import org.eclipse.smarthome.automation.annotation.ActionScope;
+import org.eclipse.smarthome.automation.annotation.RuleAction;
+import org.eclipse.smarthome.magic.binding.handler.MagicActionModuleThingHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Some automation actions to be used with a {@link MagicActionModuleThingHandler}
+ *
+ * @author Stefan Triller - initial contribution
+ *
+ */
+@ActionScope(name = "magic")
+public class MagicActionProvider implements ActionProvider {
+
+    private final Logger logger = LoggerFactory.getLogger(MagicActionProvider.class);
+
+    private final MagicActionModuleThingHandler handler;
+
+    public MagicActionProvider(MagicActionModuleThingHandler handler) {
+        this.handler = handler;
+    }
+
+    @RuleAction(label = "Stefans Thing Label", description = "Stefans Thing Description")
+    public @ActionOutput(name = "output1", type = "java.lang.String") @ActionOutput(name = "output2", type = "java.lang.String") Map<String, Object> thingHandlerAction(
+            @ActionInput(name = "input1") String input1, @ActionInput(name = "input2") String input2) {
```

Comment:
Can the input name be automatically derived from the parameter name for better readability? If there is no access to that, I'd suggest to simply use p1, p2,... as default names, if non is given.

Reply:
They are now called "p0...pn"